library/Core/src/Juvix/Core/Translate.hs

```diff
@@ -66,6 +69,7 @@ hrToIR' = \case
     pure (IR.Let π l b)
   HR.Elim e -> IR.Elim |<< hrElimToIR' e
 
+-- | @hrElimToIR'@ works the same as @hrToIR'@ but for Elims
```

Comment:
See above -- I'd point out that there is actually an asymmetry here, with `hrToIR'` pushing, and `hrElimToIR'` generating indices.